imap_processing/ultra/l0/decom_ultra.py

```diff
@@ -122,12 +123,15 @@ def process_ultra_apids(data: list, apid: int):
     sorted_packets = sort_by_time(data, "SHCOARSE")
 
     process_function = strategy_dict.get(apid)
-    decom_data = process_function(sorted_packets, defaultdict(list))
+    decom_data = process_function(sorted_packets, defaultdict(list))  # type: ignore[misc]
+    # ToDo Change, "None" not callable
```

Comment:
```suggestion
    decom_data = process_function(sorted_packets, defaultdict(list))  # type: ignore[misc]
```

I think we can keep this `ignore` in here. This error is confusing, but luckily someone else has encountered this and documented why its happening: https://cdreimer.medium.com/fixing-none-not-callable-error-message-in-mypy-3538c954549b

So it looks like this could be fixed by changing the above line to:

```python
process_function = strategy_dict.get(apid, lambda *args: False)
```

But then this results in an error in the return type 🤷🏻 . I will leave it to you if you want to try to solve that, but this seems to me like a random one-off case that can be ignored.

Reply:
I think the return error is because of the changed to how process_function was implemented. I can simply change what the function returns to reflect what the error returns.